subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/AdoptOpenJdkRemoteBinary.java

```diff
@@ -113,8 +113,17 @@ private String determineImplementation(JavaToolchainSpec spec) {
         return "hotspot";
     }
 
+    private String determineVendor(JavaToolchainSpec spec) {
+        DefaultJvmVendorSpec vendorSpec = (DefaultJvmVendorSpec) spec.getVendor().get();
+        if (vendorSpec == DefaultJvmVendorSpec.any()) {
+            return "adoptium";
```

Comment:
It is a minor optimization but it might be beneficial to take `implementation` into account too. If `J9` is involved then it is possible to end up with two copies - `semeru-openj9` for specs with an explicit vendor and `adoptium-openj9` for the rest.